src/Json.Schema.ToDotNet/DataModelGeneratorSettings.cs

```diff
@@ -82,6 +82,12 @@ public class DataModelGeneratorSettings
         /// </summary>
         public bool GenerateCloningCode { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating what C# type to generate for Json type
+        /// <code>Integer</code>.
+        /// </summary>
+        public GenerateJsonIntegerOption GenerateJsonIntegerAs { get; set; }
```

Comment:
>GenerateJsonIntegerOption [](http://example.com/codeflow?start=15&length=25)

So where's our handling for the JSON 'number' type?

We need to first review our SARIF schema to make sure we don't need a number type. Look at this one 'rank'. The specification says it's a value from 0.0 to 100.0 and provides an explicit example of 0.7 which is conclusively not an integer (in JSON, a 0 decimal value, e.g., 100.0 *can* be interpreted as an integer).

I think it's clear we'll need `generate-json-number-as`. :)

[https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317663](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317663)

Reply:
Created an issue to track: [Evaluate and add option `--generate-json-number-as`](https://github.com/microsoft/jschema/issues/160)